### PR TITLE
Add support for non-pipe webcam recordings (RecordRTC, AWS S3)

### PR DESCRIPTION
**Pull request by becky-gilbert**
Relates to issue: #1074 (but merging this PR shouldn't automatically close the issue)

This PR contains some of the changes required to support our pipe-replacement webcam recordings:
- Adds AWS S3 environment variables (to be passed into the EFP build, and used when accessing videos for lookit-api views like researcher downloads and participant study history)
- Adds new 'recording_method' field  (value: 'pipe' or 'recordrtc') to Response model
- Adds new video API view that allows AWS S3 to send information about new video uploads, which is then used to create new Video objects in the database
- Adds video API tests

Implementation notes:
- I did not create a user in the DB for authenticating the video POST requests (initiated by AWS Lambda function which is triggered in response to AWS S3 video uploads). This was because the authentication methods don't seem applicable to a user that doesn't have auth via session info. Instead I've required a hash in the header that is created using the POST message body and a secret key. I think this is as secure as using a user account with an authentication token. But I'm not an expert in this area and happy to hear feedback!
- My approach to the video POST request authentication/permission is the same as that currently used with Pipe. Pipe fires a trigger to the lookit-api when a video is uploaded to S3, and that trigger is handled by the `RenameVideoView` ([`exp/views/video.py`](https://github.com/lookit/lookit-api/blob/develop/exp/views/video.py)). The Pipe request is authenticated by creating a hash of the secret `PIPE_WEBHOOK_KEY` and the request content, and then checking that value against the one sent by Pipe in the request header (`HTTP_X_PIPE_SIGNATURE`).
- The study/response fields are removed when the HMAC signature is calculated because their format/values are different in the request object that is passed to the permissions class (vs what was actually sent by the client). But if this is a problem then it's probably possible to reconstruct the format/values sent by the client when we calculate the signature.
- I added the new 'recording_method' field to the Response table for two reasons: (1) to make it easier to retrieve videos from the correct AWS bucket whenever we need to request them (researcher downloads and consent ruling, participant's study history page), and (2) to help with any video-related debugging that might be needed while we're allowing both Pipe and non-Pipe studies to run on the platform.

### CODE REVIEW

**Review by okaycj — CHANGES_REQUESTED** [review]
Just one question for now.

File: studies/models.py [hunk 1]
```diff
@@ -1401,6 +1402,10 @@ def display_name(self):
     def download_url(self):
         return get_download_url(self.full_name)
 
+    @property
+    def recording_method(self):
+        return self.response.recording_method
```
[the remark this answers is not in the document]
Comment (becky-gilbert) [reply studies/models.py:1407]: Got it, thanks!

File: studies/models.py [hunk 2]
```diff
@@ -1401,6 +1402,14 @@ def display_name(self):
...
         return get_download_url(self.full_name)
 
+    @property
+    def recording_method_is_pipe(self):
+        return str.lower(self.response.recording_method) == "pipe"
+
+    @property
+    def recording_method_is_recordrtc(self):
+        return str.lower(self.response.recording_method) == "recordrtc"
+
```
Comment (becky-gilbert) [inline studies/models.py:1412]: @okaycj how's this?

File: studies/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (becky-gilbert) [reply studies/models.py]: I've updated this by turning this into two methods that return booleans: `recording_method_is_pipe` and `recording_method_is_recordrtc`. I flagged the change for you with [this comment](https://github.com/lookit/lookit-api/pull/1237/files#diff-990438d44609ae4095e9e633d1447b4858ec59867eaa939f30b4e3cf9cebd684R1405-R1412). Please let me know if you have feedback!

File: studies/models.py (hunk 2 above)
Comment (okaycj) [reply studies/models.py:1412]: Nice.

**Review by okaycj — APPROVED** [review]
I can't wait to test this in develop.  Nice work.

**Pull request merged**